src/ucp/core/ucp_am.c

```diff
@@ -70,21 +101,21 @@ UCS_PROFILE_FUNC(ucs_status_t, ucp_worker_set_am_handler,
 
     UCP_WORKER_THREAD_CS_ENTER_CONDITIONAL(worker);
 
-    if (id >= worker->am_cb_array_len) {
-        num_entries = ucs_align_up_pow2(id + 1, UCP_AM_CB_BLOCK_SIZE);
-        worker->am_cbs = ucs_realloc(worker->am_cbs, num_entries *
-                                     sizeof(ucp_worker_am_entry_t),
+    if (id >= worker->am.cbs_array_len) {
+        num_entries    = ucs_align_up_pow2(id + 1, UCP_AM_CB_BLOCK_SIZE);
+        worker->am.cbs = ucs_realloc(worker->am.cbs, num_entries *
```

Comment:
please, use separate pointer
and if it's not failed, i.e. return value != NULL, assign to `worker->am.cbs`
+ add check for NULL

Reply:
I guess it could memory leak in case `realloc` fails due to lack of memory to satisfy user's request.
So, better to use separate variable and if it `realloc` was okay, assign the result of `realloc` to `worker->am.cbs`

Reply:
👍, it realloc. Will fix